src/main/java/network/brightspots/rcv/ContestConfigMigration.java

```diff
@@ -17,61 +17,25 @@
 
 package network.brightspots.rcv;
 
+import static com.fasterxml.jackson.core.util.VersionUtil.parseVersion;
 import static network.brightspots.rcv.Utils.isNullOrBlank;
 
-import java.util.ArrayList;
+import com.fasterxml.jackson.core.Version;
 import java.util.Map;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
 import network.brightspots.rcv.RawContestConfig.ContestRules;
 import network.brightspots.rcv.RawContestConfig.CvrSource;
 import network.brightspots.rcv.Tabulator.TiebreakMode;
 import network.brightspots.rcv.Tabulator.WinnerElectionMode;
 
 final class ContestConfigMigration {
-
-  private static final Pattern versionNumPattern = Pattern.compile("(\\d+).*");
-
   private ContestConfigMigration() {
   }
 
-  private static ArrayList<Integer> parseVersionString(String version) {
-    ArrayList<Integer> parsed = new ArrayList<>();
-    String[] arr = version.split("\\.");
-    for (String numString : arr) {
-      Matcher m = versionNumPattern.matcher(numString);
-      if (m.matches()) {
-        parsed.add(Integer.parseInt(m.group(1)));
-      }
-    }
-
-    return parsed;
-  }
-
   // not intended to be used if either version is null
-  private static boolean isVersionNewer(String version1, String version2) {
-    boolean isNewer = false;
-    if (!version1.equals(version2)) {
-      ArrayList<Integer> version1Parsed = parseVersionString(version1);
-      ArrayList<Integer> version2Parsed = parseVersionString(version2);
-
-      for (int i = 0; i < version1Parsed.size(); i++) {
-        if (version2Parsed.size() <= i) {
-          isNewer = true;
-          break;
-        }
-        int version1Num = version1Parsed.get(i);
-        int version2Num = version2Parsed.get(i);
-        if (version1Num > version2Num) {
-          isNewer = true;
-          break;
-        } else if (version2Num > version1Num) {
-          break;
-        }
-      }
-    }
-
-    return isNewer;
+  public static boolean isVersionNewer(String version1, String version2) {
```

Comment:
This is all kind of making me wonder if `checkConfigVersionMatchesApp` is actually doing anything besides logging?? It doesn't seem like it's stopping the code from running, so I'm a bit confused by this. Would you mind giving `checkConfigVersionMatchesApp` a sanity check?

Reply:
It doesn't do anything besides log. Based on the comments, this isn't a problem in the GUI because the GUI tries to migrate versions without any care as to the existing version (which #706 should address). For now, all it does is spit out a SEVERE and continue tabulation.

Reply:
Yeah, this is kind of weird, and I wonder if this code was removed unintentionally... in the past I _think_ we had it block tabulation if the config file version was newer than the tabulator version. Does that make sense to you?

Conversely, if the config file version is older than the tabulator version, I'd think attempting to run via the CLI should block it and say "you need to migrate this via the GUI". What do you think?

@tarheel @chughes297 do either of you have any idea why we apparently don't block tabulation in either of these cases?

Reply:
This PR might be good to go, depending on what @chughes297 and / or @tarheel think about this comment.

Reply:
I'm not sure about the history, but I think I agree that 1) it should fail if the config is newer than the app and 2) it should  warn but continue and try to auto-migrate if the app is newer than the config.

Reply:
Right on, agreed. Opened #712 for this.